en/_posts/2017-01-31-kafka-metrics-grafana.md

```diff
@@ -0,0 +1,121 @@
+
+This guide will help you to understand step by step how to display your Kafka applications metrics using [Grafana](https://grafana.tsaas.ovh.com) and the [Metrics Data Platform](https://www.ovh.com/fr/data-platforms/metrics/).
+
+## Preparation
+### Get your application token to use the Metrics API
+Read the token using this link: [Token OVH API](https://api.ovh.com/console/#/dbaas/queue/%7BserviceName%7D/metrics/account#GET).
+
+![Application Token](/kb/images/2017-01-31-kafka-metrics-grafana/token.png)
+
+The token is composed by 2 parts: the user and the password. Save both somewhere safe.
+
+Note: The `serviceName` required must be your OVH Kafka application ID. You can find it in your [OVH manager](https://www.ovh.com/manager/sunrise/dbaasQueue/index.html#/dbaasQueue) when you click on the "DBaaS Queue" menu. On the left you can see the list of your applications including their IDs:
+
+![Application ID](/kb/images/2017-01-31-kafka-metrics-grafana/application_id.png)
+
+## Grafana configuration
+Open [OVH Grafana](https://grafana.tsaas.ovh.com) and log in with your OVH account.
+To configure a new datasource, you first need to create your own organization.
+
+### Create your own organization
+
+Once you're logged in, go in the top left hand corner menu (with the Grafana logo), and click on your name, then a submenu is opened, click on "+ New Organization":
+
+![New Organization](/kb/images/2017-01-31-kafka-metrics-grafana/new_organization.png)
+
+Fill the organization name that you want and create it.
+
+### Configure a datasource
+
+Now go back in the menu, you should see now an item called "Data Sources".
+
+![New Data Source](/kb/images/2017-01-31-kafka-metrics-grafana/data_source.png)
+
+Click on it.
+Then click on the "Add data source" button to configure one as below:
+
+HTTP Settings:
+
+  - Name: This is up to you
+  - Type: OpenTSDB
+  - Url: https://opentsdb-gra1.tsaas.ovh.com
+  - Access: Proxy
+  - Http Auth: Basic Auth: true
+  - With Credentials: false
+  - User: The first part of the token (before the semi-colon) that you got previously.
+  - Password: The second part of the token (after the semi-colon).
+
+OpenTSDB Settings:
+
+  - Version: ==2.2
+  - Resolution: second
+
+Then add it.
+
+### Create a dashboard
+
+Go back once again in the menu, go in "Dashboards" and click on "New".
+You should have a blank page. On the left under the menu, you should see a green menu.
+
+### Display Metrics on a Graph
+
+Open it and click on "Add Panel" -> "Graph".
+
+![New Graph](/kb/images/2017-01-31-kafka-metrics-grafana/new_graph.png)
+
+#### Graph general configuration
+
+Let's fill the "Metrics" tab, which is the most important. Firstly in "Panel data source", select the datasource that you just created:
+
+![New Graph](/kb/images/2017-01-31-kafka-metrics-grafana/select_datasource.png)
+
+Then fill the fields:
+
+  - Metric: kafka.bytes-in-per-sec.m1 or kafka.bytes-in-per-sec.m1 or
```

Comment:
Typo on the metric name (kafka.bytes-in-per-sec.m1) appears twice.
We actually expose 3 different metrics:
- kafka.bytes-in-per-sec.m1
- kafka.messages-in-per-sec.m1
- consumer.lag